langchain4j-core/src/main/java/dev/langchain4j/model/chat/observability/ChatLanguageModelListener.java

```diff
@@ -0,0 +1,48 @@
+package dev.langchain4j.model.chat.observability;
+
+import dev.langchain4j.Experimental;
+
+/**
+ * TODO
+ */
+// TODO name: ChatLanguageModelObserver?
+// TODO package
+@Experimental
+public interface ChatLanguageModelListener {
```

Comment:
What about TracingListener?

Reply:
Then ObservabilityListener

Reply:
This is a general middleware piece, I don't think it should have any ties with Observability in terms of naming.

Reply:
I will keep it as `dev.langchain4j.model.listener.ModelListener<Request, Response>`

Reply:
different types of models can have different request/response data